### ISSUE TITLE
exceptionFromStreamError fails to parse errors correctly

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#2771]
|[<img alt="ralphm's avatar" src="https://avatars.githubusercontent.com/u/116741?s=50" width="50" height="50">](https://github.com/ralphm)| @ralphm reported|
|-|-|
|Trac ID|trac#2771|
|Type|defect|
|Created|2007-08-03 11:58:36Z|

The `_parseError` function from `twisted.words.protocols.jabber.error` only works for stanza errors.

<details><summary>Searchable metadata</summary>

```
trac-id__2771 2771
type__defect defect
reporter__ralphm ralphm
priority__highest highest
milestone__Words_0_6 Words-0.6
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__words words
keywords__ 
time__1186142316000000 1186142316000000
changetime__1186476624000000 1186476624000000
version__None None
owner__ralphm ralphm
cc__ralphm cc__therve
```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="ralphm's avatar" src="https://avatars.githubusercontent.com/u/116741?s=50" width="50" height="50">](https://github.com/ralphm)<a name="note_1"></a>|@ralphm removed owner|
|-|-|

Fixed. Also cleaned up tests for this module and added docstrings.

Please review in [branches/stream-error-2771.](branches/stream-error-2771.)

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_2"></a>|@therve set owner to @ralphm|
|-|-|

Looks great please merge. One nit: please document the errorNamespace of _parseError in its docstring. Then please merge.

**Comment by twisted-trac** [comment]
|[<img alt="ralphm's avatar" src="https://avatars.githubusercontent.com/u/116741?s=50" width="50" height="50">](https://github.com/ralphm)<a name="note_3"></a>|@ralphm set status to `closed`|
|-|-|

(In [20973]) Fix t.w.p.jabber.error to properly parse XMPP stream errors.

Author: ralphm
Reviewer: therve
Fixes [#2771](https://github.com/twisted/twisted/issues/2771)
